core/src/components/cat-tabs/cat-tabs.tsx

```diff
@@ -0,0 +1,107 @@
+import { Component, h, Element, State, Watch, Listen, Host, Prop } from '@stencil/core';
+
+/**
+ * @part tab - The header of the tab.
+ */
+@Component({
+  tag: 'cat-tabs',
+  styleUrl: 'cat-tabs.scss',
+  shadow: true
+})
+export class CatTabs {
+  private tabs: HTMLCatTabElement[] = [];
+  private buttons: HTMLCatButtonElement[] = [];
+
+  @Element() hostElement!: HTMLElement;
+
+  @State() activeTabId?: string;
+
+  /**
+   * The ID of the active tab.
+   */
+  @Prop() activeTab = '';
+
+  /**
+   * The tabs alignment
+   */
+  @Prop() tabsAlign: 'left' | 'center' | 'justify' = 'left';
+
+  @Watch('activeTabId')
+  onActiveTabChanged(newActiveTab: string): void {
+    const activeTab = this.tabs.find(value => value.id === newActiveTab);
+    activeTab?.click();
+  }
+
+  componentWillLoad(): void {
+    this.tabs = Array.from(this.hostElement.querySelectorAll('cat-tab'));
+    if (this.tabs.length > 0) {
+      let nextUniqueId = 0;
```

Comment:
defining this here will lead to non-unique IDs for different tabs components. we need to push it o the outside of the component just like the other examples